framework/auth/views.py

```diff
@@ -409,7 +420,7 @@ def external_login_confirm_email_get(auth, uid, token):
             mimetype='html',
             user=user
         )
-        service_url = service_url + '?new=true'
+        service_url += '&{}'.format(urllib.urlencode({'new': 'true'}))
```

Comment:
Is `service_url` guaranteed to have at least one query parameter here? If so, please add a test that ensures not having a redirect url in this flow fails in the expected way.